src/AdaProx.jl

```diff
@@ -272,6 +272,43 @@ function stepsize(rule::OurRule, (gamma1, gamma0), x1, grad_x1, x0, grad_x0)
     return (gamma, sigma), (gamma, gamma1)
 end
 
+
+
+struct OurRulePlus{R}
+    gamma::R
+    xi::R
+    nu::R
+    r::R
+end
+
+function OurRulePlus(; gamma = 0, nu = 1, xi = 1, r = 1/2)
+    _gamma = if gamma > 0
```

Comment:
You could also avoid introducing `_gamma` and just do 

```
if gamma <= 0
    error(…)
end
```

no?

Reply:
I see.  This stepsize rule is fed into the PD algorithm. That is why I kept the same style as we had for OurRule. Only sigma is changed to gamma because we should first check how exactly the stepsize rule looks like there.